components/interactive_dialog/dialog_element.jsx

```diff
@@ -154,6 +155,17 @@ export default class DialogElement extends React.PureComponent {
                     value={this.state.value}
                 />
             );
+        } else if (type === 'bool') {
+            return (
+                <BoolSetting
+                    id={name}
+                    label={displayNameContent}
+                    value={value || false}
```

Comment:
When testing this the returned value in `model.SubmitDialogRequest.Submission` had the type `string` instead of `bool`. This is a unexpected behaviour for integrations and puts arbitrary work for type checking onto them.

I suppose this has the be fixed on the server side. @hvhallmann Would you be interested in working on this? If not, no problem. I will open a new ticket in this case for someone else.

Reply:
its weird, because when I receive the response on the API responsible for the integration, I can see a boolean value;
![Captura de tela de 2019-07-17 00-17-49](https://user-images.githubusercontent.com/7022648/61344917-60a40c00-a829-11e9-89ce-8f7e39d903d1.png)

but I believe you are talking about some code located at mattermost-server basecode, right?
If yes, please open a new ticket, I don't have experience on that part;

Reply:
I will open a ticket for this.

Reply:
Done, see https://mattermost.atlassian.net/browse/MM-17519